Purchases/Public/RCPurchases.m

```diff
@@ -953,8 +953,11 @@ - (void)handleReceiptPostWithTransaction:(SKPaymentTransaction *)transaction
         [self markAttributesAsSyncedIfNeeded:subscriberAttributes appUserID:self.appUserID error:error];
 
         RCPurchaseCompletedBlock completion = nil;
+        NSString *productIdentifier = transaction.payment.productIdentifier;
```

Comment:
let's add nullable since we're going to check for nullabiilty. 
we should also maybe add a log entry, in case a future dev looks at the docs and (reasonably) assumes that this should never be nil.

Reply:
we could have a method that unwraps the whole thing and RCLogs if we're using it in a couple of places.

Reply:
I added it in d735da0, I doubted about making an extension function on `SKTransaction`, what do you think?